EventsExpress/Mapping/EventMapperProfile.cs

```diff
@@ -104,49 +98,100 @@ public EventMapperProfile()
                 .ForMember(dest => dest.Visitors, opts => opts.Ignore());
         }
 
+        private static LocationDto MapLocation(Event e)
+        {
+            return new LocationDto()
+            {
+                Id = e.Id,
+                OnlineMeeting = e.EventLocation.OnlineMeeting,
+                Point = e.EventLocation.Point,
+                Type = e.EventLocation.Type,
+            };
+        }
+
+        private static LocationDto MapLocation(EventEditViewModel e)
```

Comment:
Could we use polymorphism here to avoid code duplication?

Reply:
Maybe we could, but at this moment i don't know how i can exactly to it. So, to avoid code dublication i used a new generic class which has a generic method to map to LocationDto
`    public static class MapToLocationDto<T>
    {
        public delegate Point PointAction(T e);

        public static LocationDto Map(T eventViewModel, PointAction pointAction)
        {
            dynamic e;
            if (typeof(T) == typeof(EventCreateViewModel))
            {
                e = eventViewModel as EventCreateViewModel;
            }
            else if (typeof(T) == typeof(EventEditViewModel))
            {
                e = eventViewModel as EventEditViewModel;
            }
            else
            {
                throw new EventsExpressException("Cannot cast generic T to any type of Event View Model in Map to Location Dto");
            }

            var locationDto = new LocationDto();
            if (e.Location != null)
            {
                if (e.Location.Type != null)
                {
                    locationDto.Type = e.Location.Type;
                }

                if (e.Location.OnlineMeeting != null && e.Location.Type == LocationType.Online)
                {
                    locationDto.OnlineMeeting = e.Location.OnlineMeeting;
                }

                if (e.Location.Type == LocationType.Map)
                {
                    locationDto.Point = pointAction.Invoke(e);
                }
            }

            return locationDto;
        }
    }`